### PR TITLE
Always offline

### PR DESCRIPTION
**Pull request by grahamgilchrist**
Initially this library only applies offline functionality to `$http` request which explicitly include the `offline: true` config flag.

When you want to cache the majority of requests in your app, including those made by libraries, this is not always possible and quite annoying to add to every request.

This PR adds an option to enable offline functionality for all requests by default, with an exclude list option to exclude specific cache IDs.

Request is offlined if:
- `offline === true`
- `alwaysOffline === true` and `config.offline !== false` and cache id not in exclude list

### CODE REVIEW

**Comment by grahamgilchrist** [comment]
@incuna/js Please review/merge

File: angular-offline.js [hunk 1]
```diff
@@ -79,15 +102,27 @@ angular
...
 
     /**
+     * Return cache/default cache
+     *
+     * @param {object} cache Cache
+     * @returns {object} Cache
+     */
+
+    function getCache (cache) {
+      if (cache === true)
```
Comment (wytrych) [inline angular-offline.js:112]: Add `{}`

File: README.md [hunk 2]
```diff
@@ -57,6 +57,29 @@ module.config(function (offlineProvider) {
 });
 ```
 
+### offlineProvider.alwaysOnline(value)
+
+Enable or disable applying offline functionality to all requests. When this is false, requests will only be offlined by explicitly adding the `offline: false` property to the request config. If you want to enable offline functionality for the majority or all of the reqeusts in the application, adding the offline flag to every request is not ideal. Setting `alwaysOnline` to `true`, means that offline functionality will be applied to every request, except those that explciitly set `offline: false` in their individual request configurations, or caches that are excluded (see excludeCacheIds).
```
Comment (kevinetienne) [inline README.md]: `reqeusts` - > `requests`
`explciitly` -> `explicitly`

File: angular-offline.js (hunk 1 above)
Comment (wytrych) [reply angular-offline.js:112]: Or we don't do our stile since it's an external repo?

**Comment by wytrych** [comment]
Would you try to PR it back in to the orignal repo?

File: angular-offline.js (hunk 1 above)
Comment (grahamgilchrist) [reply angular-offline.js:112]: Well it was forked from an external lib which is no longer maintained, so it doesn't have our style rules applied.

File: README.md (hunk 2 above)
Comment (grahamgilchrist) [reply README.md]: Fixed :)

**Comment by grahamgilchrist** [comment]
@wytrych Sadly not. The original is unmaintained and never going to get an update. Although a fork, this is essentially a library we are now maintaining separately.

**Pull request merged**